src/cache/file_cache.cc

```diff
@@ -115,33 +126,60 @@ using namespace string;
 // static
 HandledHash FileCache::Hash(HandledSource code, CommandLine command_line,
                             Version version) {
-  return HandledHash(base::Hexify(code.str.Hash()) + "-" +
+  return FileCache::Hash(code, List<Immutable>{}, command_line, version);
+}
+
+// static
+HandledHash FileCache::Hash(HandledSource code,
+                            const List<Immutable>& extra_files,
+                            CommandLine command_line, Version version) {
+  return HandledHash(HashCombine(code.str, extra_files) + "-" +
                      base::Hexify(command_line.str.Hash(4)) + "-" +
                      base::Hexify(version.str.Hash(4)));
 }
 
 // static
 UnhandledHash FileCache::Hash(UnhandledSource code, CommandLine command_line,
                               Version version) {
+  return FileCache::Hash(code, List<Immutable>{}, command_line, version);
+}
+
+// static
+UnhandledHash FileCache::Hash(UnhandledSource code,
```

Comment:
It's better to leave a single Hash method: either put the extra_files in the end so they can be set to empty default, or simply pass empty brace-initializer everywhere throughout the code.